### DISCUSSION TITLE
Best way to sync / ingegrate Saleor with Facebook Market Place

### DISCUSSION THREAD

**Discussion by IsaacRead** [saleor/apps#316] [Q&A] [answered]
Hey there,

We are developing a store using Saleor, but we also want to have the products that we add to Saleor automatically synced with our Facebook shop.
I was thinking of creating a Saleor App and using a webhook to listen to changes to the catalouge and then use the Facebook api to update the facebook page based on those changes.
Am I headed in the right direction?
Has anyone done this before?
Any help / pointers greatly appreciated

Cheers

**Comment by lkostrowski** [comment, marked as answer]
Hey @IsaacRead 

We are currently working on the official Saleor App - Product Feed
You can find the source code [here](https://github.com/saleor/apps/tree/main/apps/products-feed)

In the first scope it will generate the XML for Google Merchant, but you should be able to convert it easily.

Current solution is not using webhooks - it exposes an endpoint that Google can call and it will generate feed on the fly, using the API.

However this will not work for bigger datasets - in such scenarios App will have to listed to product changes and regenerate XML file.

App will be available to install in App Store (Saleor Cloud) in few weeks, but you can check out the code, deploy it on yourself or just learn how its done. Also feel free to contribute 🚀

**Comment by fedexin40** [comment]
@IsaacRead what was your solution for this?

**Reply by simon-online** [reply]
We ended up just building a seperate script that fetches the products from the API and creates the Facebook product CSV file on a URL that we then configure with a scheduled feed upload:

https://en-gb.facebook.com/business/help/120325381656392?id=725943027795860

This will work well with a large number of products.
